experimental/devices/inky/inky.go

```diff
@@ -0,0 +1,327 @@
+// Copyright 2019 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package inky
+
+import (
+	"encoding/binary"
+	"fmt"
+	"image"
+	"image/color"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/display"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/physic"
+	"periph.io/x/periph/conn/spi"
+)
+
+const (
+	// Constants for an Inky pHAT
+	cols = 104
+	rows = 212
+)
+
+// Color is used to define which model of inky is being used, and also for
+// setting the border color.
+type Color int
+
+const (
+	Black  = iota
+	Red    = iota
+	Yellow = iota
+	White  = iota
+)
+
+var borderColor = map[Color]byte{
+	Black:  0x00,
+	Red:    0x33,
+	Yellow: 0x33,
+	White:  0xff,
+}
+
+// NewpHAT opens a handle to an Inky pHAT.
+func NewpHAT(p spi.Port, dc gpio.PinOut, reset gpio.PinOut, busy gpio.PinIn, color Color) (*Dev, error) {
+	if color != Black && color != Red && color != Yellow {
```

Comment:
I'm thinking we may group all the pimonori drivers into a subpackage. I'll think about it.

Another thing is that generally I prefer a `New()` function. Here do you plan to add another New function? Otherwise I'd prefer to rename.

Reply:
I called it `NewpHAT()` to leave room for a `NewwHAT()` later.

Reply:
It looks like from the python code that only effective difference is the dimensions, so maybe pass this as an Opts struct? e.g.

```
// Opts is the options to specify which device is being controlled, and
// its default settings.
type Opts struct {
  // Model being used.
  Model Model
  // Model color used.
  ModelColor Color.
  // Initial border color. Will be set on the first Draw().
  Border Color
}

// Model lists the supported e-ink display models.
type Model int

const (
  PHAT Model = iota
  WHAT
)

// New opens a handle to an Pimonori Inky pHAT or wHAT.
func New(p spi.Port, dc gpio.PinOut, reset gpio.PinOut, busy gpio.PinIn, o *Opts) (*Dev, error) {
  ...
}
```

wdyt?

Reply:
Yes, that would work too.